tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -721,6 +720,217 @@ void TestCase(ClientEncryption testCaseClientEncription, BsonDocument masterKey,
             }
         }
 
+        [SkippableTheory]
+        [MemberData(nameof(DeadlockTest_MemberData))]
+        public void DeadlockTest(
+            string _,
+            int maxPoolSize,
+            bool bypassAutoEncryption,
+            string keyVaultMongoClientKey,
+            int expectedNumberOfClients,
+            string[] clientEncryptedEventsExpectation,
+            string[] clientKeyVaultEventsExpectation,
+            bool async)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+            RequirePlatform
+                .Check()
+                .SkipWhen(SupportedOperatingSystem.Linux, SupportedTargetFramework.NetStandard15)
+                .SkipWhen(SupportedOperatingSystem.MacOS, SupportedTargetFramework.NetStandard15);
+
+            var clientKeyVaultEventCapturer = CreateEventCapturer();
+            using (var client_keyvault = CreateMongoClient(maxPoolSize: 1, writeConcern: WriteConcern.WMajority, readConcern: ReadConcern.Majority, eventCapturer: clientKeyVaultEventCapturer))
+            using (var client_test = ConfigureClient(clearCollections: true, writeConcern: WriteConcern.WMajority, readConcern: ReadConcern.Majority))
+            {
+                var dataKeys = GetCollection(client_test, __keyVaultCollectionNamespace);
+                var externalKey = JsonFileReader.Instance.Documents["external.external-key.json"];
+                Insert(dataKeys, async, externalKey);
+
+                var externalSchema = JsonFileReader.Instance.Documents["external.external-schema.json"];
+                CreateCollection(client_test, __collCollectionNamespace, new BsonDocument("$jsonSchema", externalSchema));
+
+                using (var client_encryption = ConfigureClientEncryption(client_test.Wrapped as MongoClient, kmsProviderFilter: "local"))
+                {
+                    var value = "string0";
+                    var encryptionOptions = new EncryptOptions(
+                        algorithm: EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic.ToString(),
+                        alternateKeyName: "local");
+                    var ciphertext = ExplicitEncrypt(client_encryption, encryptionOptions, value, async);
+
+                    var eventCapturer = CreateEventCapturer().Capture<ClusterOpeningEvent>();
+
+                    using (var client_encrypted = ConfigureClientEncrypted(
+                        kmsProviderFilter: "local",
+                        maxPoolSize: maxPoolSize,
+                        bypassAutoEncryption: bypassAutoEncryption,
+                        eventCapturer: eventCapturer,
+                        externalKeyVaultClient: GetKeyVaultMongoClientByKey()))
```

Comment:
Local method `GetKeyVaultMongoClientByKey` is redundant. Consider:
```suggestion
                        externalKeyVaultClient: keyVaultMongoClientKey == "client_keyvault" ? client_keyvault : null))
```

Reply:
mostly agreed, I did this because the spec test idea was to map a provided string key to the client, currently, we have only one test key, but I think it's a valid assumption that the number of keys can be bigger further, so I would not change this